test/__snapshots__/test.js.snap

```diff
@@ -4558,6 +4558,129 @@ const Impl = require(\\"../implementations/URL.js\\");
 "
 `;
 
+exports[`URLCallback.webidl 1`] = `
+"\\"use strict\\";
+
+const conversions = require(\\"webidl-conversions\\");
+const utils = require(\\"./utils.js\\");
+
+module.exports = {
+  convert(value, { context = \\"The provided value\\" } = {}) {
+    if (typeof value !== \\"function\\" && !treatNonObjectAsNull) {
+      throw new TypeError(context + \\" is not a function\\");
+    }
+
+    function wrapper(...args) {
+      if (new.target !== undefined) {
+        throw new Error(
+          \\"Internal error: do not call converted callback function with \`new\`; instead use \\" +
+            \\"convertedFunc.construct()\\"
+        );
+      }
+
+      let returned;
+
+      for (let i = 0; i < args.length; i++) {
+        args[i] = utils.tryWrapperForImpl(args[i]);
+      }
+      if (args.length < 2) {
```

Comment:
Reusing the args array is a bit confusing, hmm. I guess it's a bit more efficient though, so I don't know.

Also, I think this code could be simplified as follows:

```js
for (let i = args.length; i < 2; ++i) {
  args[i] = undefined;
}
args.length = 2;
```

And I mean, ideally the generated code would look something like

```js
const convertedArgs = [
  0 in args ? utils.tryWrapperForImpl(args[0]) : undefined,
  1 in args ? utils.tryWrapperForImpl(args[1]) : undefined
];
```

but that level of tuning might not be worth while, I dunno.

Reply:
This (admitted not great) algorithm came from https://heycam.github.io/webidl/#web-idl-arguments-list-converting, which has something that looks like this. I wouldn't be opposed to something that looks like your second option above though, if the complexity is manageable.

Reply:
I guess it could also be in utils.js, so that the call site would then just look like `convertArgsList(inputArgs, 2)`.

Reply:
Eventually there might be more complicated conversion from IDL values to ES values (#71). https://heycam.github.io/webidl/#web-idl-arguments-list-converting takes a Web IDL arguments list, so I'd like to keep it that way.

Reply:
Oh wait forget about that. It's not actually dependent on the types.

Reply:
Always setting `args.length` to a constant is only valid for callbacks without optional arguments.